validator/client/src/test/java/tech/pegasys/teku/validator/client/metrics/MetricRecordingValidatorApiChannelTest.java

```diff
@@ -0,0 +1,189 @@
+        noResponseTest(
+            "subscribeToPersistentSubnets",
+            channel -> channel.subscribeToPersistentSubnets(emptySet()),
+            MetricRecordingValidatorApiChannel.PERSISTENT_SUBSCRIPTION_COUNTER_NAME),
+        noResponseTest(
+            "sendSignedAttestation",
+            channel -> channel.sendSignedAttestation(dataStructureUtil.randomAttestation()),
+            MetricRecordingValidatorApiChannel.PUBLISHED_ATTESTATION_COUNTER_NAME),
+        noResponseTest(
+            "sendAggregateAndProof",
+            channel ->
+                channel.sendAggregateAndProof(dataStructureUtil.randomSignedAggregateAndProof()),
+            MetricRecordingValidatorApiChannel.PUBLISHED_AGGREGATE_COUNTER_NAME),
+        noResponseTest(
+            "sendSignedBlock",
+            channel -> channel.sendSignedBlock(dataStructureUtil.randomSignedBeaconBlock(1)),
+            MetricRecordingValidatorApiChannel.PUBLISHED_BLOCK_COUNTER_NAME));
+  }
+
+  private static Arguments noResponseTest(
+      final String name, final Consumer<ValidatorApiChannel> method, final String counterName) {
+    return Arguments.of(name, method, counterName);
+  }
+
+  public static Stream<Arguments> getDataRequestArguments() {
+    final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+    final UnsignedLong slot = dataStructureUtil.randomUnsignedLong();
+    final BLSSignature signature = dataStructureUtil.randomSignature();
+    final AttestationData attestationData = dataStructureUtil.randomAttestationData();
+    return Stream.of(
+        requestDataTest(
+            "getForkInfo",
+            ValidatorApiChannel::getForkInfo,
+            MetricRecordingValidatorApiChannel.FORK_REQUESTS_COUNTER_NAME,
+            dataStructureUtil.randomForkInfo()),
+        MetricRecordingValidatorApiChannelTest.requestDataTest(
+            "getDuties",
+            channel -> channel.getDuties(slot, Collections.emptyList()),
+            MetricRecordingValidatorApiChannel.DUTIES_REQUESTS_COUNTER_NAME,
+            Collections.emptyList()),
+        requestDataTest(
+            "createUnsignedBlock",
+            channel -> channel.createUnsignedBlock(slot, signature),
+            MetricRecordingValidatorApiChannel.UNSIGNED_BLOCK_REQUESTS_COUNTER_NAME,
+            dataStructureUtil.randomBeaconBlock(slot)),
+        requestDataTest(
+            "createUnsignedAttestation",
+            channel -> channel.createUnsignedAttestation(slot, 4),
+            MetricRecordingValidatorApiChannel.UNSIGNED_ATTESTATION_REQUEST_COUNTER_NAME,
+            dataStructureUtil.randomAttestation()),
+        requestDataTest(
+            "createAggregate",
+            channel -> channel.createAggregate(attestationData),
+            MetricRecordingValidatorApiChannel.AGGREGATE_REQUESTS_COUNTER_NAME,
+            dataStructureUtil.randomAttestation()));
+  }
+
+  private static <T> Arguments requestDataTest(
```

Comment:
nit: (or requestTestData)
```suggestion
  private static <T> Arguments createTestArguments(
```